lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,29 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "parameter MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
+      merged_options = merge_options(options, other.options)
+
+      IncludeDirective.new(hash, merged_options)
+    end
+
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge!(other)
+      merge_result = merge(other)
+      @hash = merge_result.hash
+      @options = merge_result.options
```

Comment:
After giving a thought about the options merging thing, I believe the most sane thing to do here is to assume that when one merges an `IncludeDirective` `B` into an other `IncludeDirective` `A`, we keep `A`'s options.